test/utils/wasm_engine.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace fizzy::test
+{
+/// The stateful representation of a wasm engine for testing purposes.
+class WasmEngine
+{
+public:
+    struct Result
+    {
+        bool trapped = false;
+        std::optional<uint64_t> value;
+    };
+
+    virtual ~WasmEngine() noexcept = default;
+
+    /// Parses input bytes and creates internal wasm module out of it.
+    /// Consecutive invocations replace the internal module and invalidate the internal instance.
+    virtual void parse(bytes_view input) = 0;
```

Comment:
Is this different to `fizzy::bytes_view`? If not I'd prefer the explicit path to match the rest of the header.

Reply:
@chfast ping

Reply:
It's the same. It should rather be `bytes_view` in other places.

Reply:
Are you fixing the others then?

Reply:
Changed.